CRM/Prospect/Hook/APIWrappers/CaseCreationAPIWrapper.php

```diff
@@ -7,23 +7,37 @@ class CRM_Prospect_Hook_APIWrappers_CaseCreationAPIWrapper {
 
   /**
    * Hooks run function.
+   *
+   * Applies the API Wrapper to update Financial Information Custom Fields.
+   *
+   * @param array $wrappers
+   *   Wrappers.
+   * @param array $apiRequest
+   *   Api Request.
    */
-  public function run(&$wrappers, $apiRequest) {
+  public function run(array &$wrappers, array $apiRequest) {
     if (!$this->shouldRun($apiRequest)) {
       return;
     }
 
-    $wrappers[] = new CRM_Prospect_APIWrapper_prospectFinancialInformationCustomFields();
+    $wrappers[] = new CRM_Prospect_APIWrapper_ProspectFinancialInformationCustomFields();
   }
 
   /**
    * Determines if the hook will run.
+   *
+   * @param array $apiRequest
+   *   Api Request.
+   *
+   * @return bool
+   *   If the hook should run.
    */
-  public function shouldRun($apiRequest) {
-    if (!($apiRequest['entity'] === 'Case' && in_array($apiRequest['action'], ['create', 'edit']))) {
-      return;
-    }
-    elseif (CRM_Prospect_Helper_ProspectHelper::isApiCallProspectCategory($apiRequest['params']['case_type_id'])) {
+  public function shouldRun(array $apiRequest) {
```

Comment:
I think something is missing here, So we don't get the condition mixed up, can you replace with the code I suggested earlier?

This here means that if `$entityIsCaseAndActionCreateOrEdit`, then the condition should run, meanwhile previous code says otherwise, Note the `NOT ` sign in earlier condition.